apps/aecore/src/aec_mining.erl

```diff
@@ -89,18 +92,19 @@ create_coinbase_tx() ->
             Error
     end.
 
--spec adjust_target(block()) -> block().
-adjust_target(Block) ->
+-spec adjust_target(block(), list(header())) ->
+            {ok, block(), aec_pow:nonce()} | {error, term()}.
+adjust_target(Block, AdjHeaders) ->
     Header = aec_blocks:to_header(Block),
-    case aec_target:determine_delta_header_height(Header) of
-        {ok, InitialHeaderHeight} ->
-            %% For mining we can grab the header at height=DeltaHeight from
-            %% the current main chain, hence aec_conductor:get_header_by_height/1 call.
-            %% TODO: It is not good to call back to aec_conductor
-            %%       but we do this for now.
-            {ok, InitialHeader} = aec_conductor:get_header_by_height(InitialHeaderHeight),
-            CalculatedTarget = aec_target:recalculate(Header, InitialHeader),
-            aec_blocks:set_target(Block, CalculatedTarget);
-        {error, chain_too_short_to_recalculate_target} ->
-            Block
+    DeltaHeight = aec_governance:blocks_to_check_difficulty_count(),
+    case aec_headers:height(Header) =< DeltaHeight of
+        true ->
+            %% For the first DeltaHeight blocks, use pre-defined target
+            {ok, Block, aec_pow:pick_nonce()};
+        false when DeltaHeight == length(AdjHeaders) ->
+            CalculatedTarget = aec_target:recalculate(Header, AdjHeaders),
+            Block1 = aec_blocks:set_target(Block, CalculatedTarget),
+            {ok, Block1, aec_pow:pick_nonce()};
+        false -> %% Wrong number of headers in AdjHeaders...
```

Comment:
I see this does not create candidate, and the caller (aec_conductor) logs error. Thanks.